contracts/svm/testing/tests/swap.rs

```diff
@@ -846,6 +846,58 @@ fn test_swap_wrong_mint_fee() {
     );
 }
 
+#[test]
+fn test_swap_wrong_token_program_fee() {
+    let SwapSetupResult {
+        mut svm,
+        user,
+        searcher,
+        token_searcher,
+        token_user,
+        router_ta_mint_searcher,
+        relayer_signer,
+        ..
+    } = setup_swap(Default::default());
+
+    let express_relay_metadata = get_express_relay_metadata(&mut svm);
+
+    let swap_args = SwapArgs {
+        deadline:         svm.get_sysvar::<Clock>().unix_timestamp,
+        amount_searcher:  token_searcher.get_amount_with_decimals(1.),
+        amount_user:      token_user.get_amount_with_decimals(1.),
+        referral_fee_bps: 1500,
+        fee_token:        FeeToken::User,
+    };
+
+    let instructions = build_swap_instructions(SwapParams {
+        searcher: searcher.pubkey(),
+        user: user.pubkey(),
+        router_fee_receiver_ta: router_ta_mint_searcher,
+        fee_receiver_relayer: express_relay_metadata.fee_receiver_relayer,
+        token_searcher: token_searcher.clone(),
+        token_user: token_user.clone(),
+        swap_args,
+        overrides: SwapParamOverride {
+            token_program_fee: Some(token_user.token_program), // wrong token program fee
+            ..Default::default()
+        },
+        relayer_signer: relayer_signer.pubkey(),
+    });
+
+    let result = submit_transaction(
+        &mut svm,
+        &instructions,
+        &searcher,
+        &[&searcher, &user, &relayer_signer],
+    )
+    .unwrap_err();
+    assert_custom_error(
+        result.err,
+        4,
+        InstructionError::Custom(AnchorErrorCode::ConstraintTokenMint.into()),
```

Comment:
wait i thought we were supposed to get `ConstraintRaw`

Reply:
true